treee111/wahooMapsCreator
# [FEAT] Integrate contour lines (elevation) into generated maps with argument `-con`

## This PR…

- closes #145 
- makes relevant changes to be able to integrate contour lines (elevantion) into the generated maps
  - python coding
  - quickstart documenation
  - usage documentation
  - tag-wahoo-poi.xml to have contour lines included in the generated maps
  - anaconda env .yml files to have the relevant tools included

## Considerations and implementations

- Most of the initial solution been described here: https://github.com/treee111/wahooMapsCreator/issues/145#issuecomment-1435660409
- A important thing is that python 3.10 is needed because matplotlib-3.4.3 is needed, later versions which work with python 3.11 do not work a la "_contour" has not been found

## How to test

1. setup a Anaconda environment with python 3.10 if you don't have one already
https://github.com/treee111/wahooMapsCreator/blob/contour-lines/docs/QUICKSTART_ANACONDA.md#create-anaconda-environment
make sure that shapely is installed via conda-forge as written in this link!

3. install additional programs
https://github.com/treee111/wahooMapsCreator/blob/contour-lines/docs/QUICKSTART_ANACONDA.md#additional-programs-for-generating-contour-lines
4. use branch `contour-lines` or alpha version installed via
~~`pip install wahoomc==4.0.0a3`~~
~~`pip install wahoomc==4.0.0a5`~~
`pip install wahoomc==4.0.0a10` or
`pip install wahoomc==4.0.0a8`
6. run `python -m wahoomc cli -co malta -con` or similar for your country / xy-coordinates

## Pull Request Checklist
    + Especially the [Pull-Requests](https://github.com/treee111/wahooMapsCreator/blob/develop/docs/CONTRIBUTING.md#Pull-Requests) section
